### ISSUE TITLE
Training loss becomes NaN during basecaller training (rna004 demo datasets, rna004_130bps_sup@v5.2.0 config )

### ISSUE THREAD

**Issue by sparkcyf** [nanoporetech/bonito#420]
Hi Bonito team,

I'm encountering an issue where the training loss becomes `NaN` when training a basecaller model using the official example dataset. The training starts, but the loss turns to `NaN` within the first epoch and remains `NaN`, resulting in 0% accuracy.

I suspect this might be related to the newer model architectures, as I've found that older models train successfully.

------

#### **Environment and Data**

To ensure reproducibility, I used the official model configurations and example data provided by ONT.

| **Item**         | **Version / Link**                                           |
| ---------------- | ------------------------------------------------------------ |
| **Model Config** | `rna004_130bps_sup@v5.2.0`     (https://cdn.oxfordnanoportal.com/software/analysis/bonito/rna004_130bps_sup@v5.2.0.zip)                              |
| **Dataset**      | `example_data_rna004_v0`          (https://cdn.oxfordnanoportal.com/software/analysis/bonito/example_data_rna004_v0.zip)                           |
| **GPU**          | NVIDIA RTX 3090                                              |
| **Software**     | bonito `0.9.1`, Python `3.13`, torch `2.6.0+cu124`, flash-attn `2.7.4.post1` |

------

#### **Steps to Reproduce**

1. Downloaded and extracted the model and dataset from the official links.

2. Ran the following training command:

   Bash

   ```
   bonito train \
   --device cuda:0 \
   --batch 48 \
   --config rna004_130bps_sup@v5.2.0/config.toml \
   --chunks 990000 \
   --valid-chunks 10000 \
   --directory example_data_rna004_v0 \
   example_data_rna004_v0_sup_5.2.0
   ```

------

#### **Observed Behavior**

The training loss immediately becomes `NaN` and never recovers.

**Training Output:**

```
[epoch 1] directory=example_data_rna004_v0_sup_5.2.0 loss=nan mean_acc=0.000% median_acc=0.000%
[990000/990000]: 100%|#########################################################| [1:30:29, loss=nan]
[epoch 2] directory=example_data_rna004_v0_sup_5.2.0 loss=nan mean_acc=0.000% median_acc=0.000%
...
```

The `losses.csv` file and the corresponding plot confirm this behavior, showing the loss flatlining at `NaN` from the beginning.

losses_1.csv:

[losses_1.csv](https://github.com/user-attachments/files/20743483/losses_1.csv)

**Loss Curve:**

![Image](https://github.com/user-attachments/assets/a05e305a-8aac-4561-9d45-f87d8269e6f6)

------

#### Additional Observations

I've also tested other model versions with the same data and environment:

- `rna004_130bps_sup@v5.0.0`: **Fails** with the same `NaN` loss issue.
- `rna004_130bps_sup@v3.0.1`: **Succeeds**, and the training proceeds as expected.

This suggests the issue may be linked to the transformer-based architecture introduced in the v5 models.

Could you please provide the set of parameters that were used to successfully train the `rna004_130bps_sup@v5.2.0` model with the `example_data_rna004_v0` dataset? This would be very helpful for troubleshooting.

Thank you!

**Comment by davidnewman02** [comment]
The `sup@v5.x` models are using a transformer architecture and these requires some different hyper-parameter tuning. I would recommend using a significantly lower learning rate if finetuning from these models, perhaps in the order of 1e-5 to 1e-6.
